sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -673,6 +673,12 @@ Exceptions:
 
 * Throws synchronously with error code `invalid` if the graph contains a
   node which targets a device not present in `syclContext`.
+  
+* Throws synchronously with error code `invalid` if the graph contains
+  any isolated node which is not reachable from all other nodes.
+  
+* Throws synchronously with error code `invalid` if the graph is empty, 
+  i.e. contains no nodes or only empty nodes.  
```

Comment:
Although I can't think of a use-case for an empty graph, I think this behaviour might be inconsistent with other APIs.
Lower level APIs let you have empty command-buffers & command-lists, and I can't see any errors specified in CUDA graphs for instantiating an empty graph, though I may be missing that wording. SYCL also lets you submit empty command-groups, so submitting an empty graph is analogous to that.

By adding this error, we do also need to remove this wording in the description of `finalize()` (line 658):

> It is valid to finalize an empty graph instance with no recorded commands.

Reply:
You're right CUDA Graphs allows to define an empty graph. But some frameworks that are using it, error in case the graph is empty. As a kind of sanity check (e.g. nbr. of nodes equals 0). Just wanted to make sure we considered both options with this PR. Will take it out for now, and we'll leave this part as is.